apps/admin-page/src/app/pages/dashboard/dashboard.component.ts

```diff
@@ -1,10 +1,454 @@
+      this.threePastMonthOrders = lastPeriodOrderCount;
+    } else {
+      this.sixPastMonthOrders = lastPeriodOrderCount;
+    }
+    return this.getPercentDiff(thisPeriodOrderCount, lastPeriodOrderCount);
+  }
+
+  getPeriodRevenue(period: number) {
+    const thisPeriodOrders = this.getPeriodOrders(period);
+    let thisPeriodRevenue = 0;
+    if (period === 14) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.twoWeekRevenue = thisPeriodRevenue;
+    } else if (period === 30) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.monthRevenue = thisPeriodRevenue;
+    } else if (period === 90) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.threeMonthRevenue = thisPeriodRevenue;
+    } else {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.sixMonthRevenue = thisPeriodRevenue;
+    }
+    return thisPeriodRevenue;
+  }
+
+  calculatePeriodRevenueDifference(periodStart: number, periodEnd: number) {
+    const thisPeriodOrders = this.getPeriodOrders(periodStart);
+    const lastPeriodOrders = this.getPreviousPeriodOrders(periodStart, periodEnd);
+    let thisPeriodRevenue = 0;
+    let lastPeriodRevenue = 0;
+
+    if (periodStart === 14 && periodEnd === 28) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      lastPeriodOrders.forEach((order) => {
+        lastPeriodRevenue += order.total;
+      });
+      this.twoPastWeekRevenue = lastPeriodRevenue;
+    } else if (periodStart === 30 && periodEnd === 60) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      lastPeriodOrders.forEach((order) => {
+        lastPeriodRevenue += order.total;
+      });
+      this.monthPastRevenue = lastPeriodRevenue;
+    } else if (periodStart === 90 && periodEnd === 180) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      lastPeriodOrders.forEach((order) => {
+        lastPeriodRevenue += order.total;
+      });
+      this.threePastMonthRevenue = lastPeriodRevenue;
+    } else {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      lastPeriodOrders.forEach((order) => {
+        lastPeriodRevenue += order.total;
+      });
+      this.sixPastMonthRevenue = lastPeriodRevenue;
+    }
+    return this.getPercentDiff(thisPeriodRevenue, lastPeriodRevenue);
+  }
+
+  // Surplus = Total - production cost - (trees planted > 1) - shipping
+  getPeriodSurplus(period: number) {
+    const thisPeriodOrders = this.getPeriodOrders(period);
+    let surplus = this.getPeriodRevenue(period);
+
+    if (period === 14) {
+      thisPeriodOrders.forEach((order) => {
```

Comment:
Same here, take the forEach out of the if statement and make this into a switch (period) {}